src/lib/Libnet/net_server.c

```diff
@@ -549,26 +608,20 @@ wait_request(time_t waittime)
 				}
 			}
 #endif
-			idx = connection_find_actual_index(em_fd);
-			if (idx < 0) {
-				return -1;
+			j = 0;
+			if (scks && scks->active_socks && prio_sock_processed) {
+				for (j = 0; j < scks->active_socks; j++) {
+                                	if (scks->socket_fd[j] == em_fd) {
+						break;
+                                	}
+                        	}
 			}
 
-			svr_conn[idx]->cn_lasttime = time(NULL);
-
-			if (svr_conn[idx]->cn_active != Primary && svr_conn[idx]->cn_active != RppComm && svr_conn[idx]->cn_active
-					!= Secondary) {
-
-				if (!(svr_conn[idx]->cn_authen & PBS_NET_CONN_AUTHENTICATED)) {
-
-					if (engage_authentication(svr_conn[idx]) == -1) {
-						close_conn(em_fd);
-						continue;
-					}
-				}
+			if (scks && (j < scks->active_socks) && prio_sock_processed) {
```

Comment:
Here also you can change the condition as follows.
if (scks && prio_sock_processed &&(j < scks->active_socks))

Reply:
done